test/jani_generator/_test_data/uc2_assembly/Plugins/IsBlockFell.scxml

```diff
@@ -21,15 +21,7 @@
             <ros_service_send_request name="get_block_status">
             </ros_service_send_request>
         </onentry>
-        <transition target="wait_answer" />
```

Comment:
The main issue lies on this bit: we expect the state to transition right away from "send_req" after requesting the service, and then start waiting for the service response in "wait_answer".
The main issue here is that it could happen, in a very unlikely case, that the service provides an answer before the transition to the new state is done, and this causes the response to be lost (because we auto-generate transitions for each state, to prevent deadlocks).
This problem is indeed solved by removing a state, but of course we need to handle this more systematically by changing the AS2FM conversion code. We have some options here:
* Determine which events should lead to auto-generated self-loops and which not (hard, a lot of special casing, not a big fan of this)
*  Prevent users to add those empty transitions, because they can be simply put together (also here, I am not too keen on adding constraints)
* Do not generate self-loops in case empty transitions are found in a state (and make sure there are no event-based transitions in the same state (a combination of the first two solutions, but I think it is a good-enough middle ground, and I would implement this in a later PR).

By the way, we could discuss this more precisely later on :)